talisker/request_id.py

```diff
@@ -53,8 +53,12 @@ def get():
 
 def set(id):
     """Sets id in both general request context, and specific logging dict"""
-    request_context.request_id = id
-    set_logging_context(request_id=id)
+    if id is None:
```

Comment:
Setting to None and deleting feel like different operations but I'm fine if you're sure they're equivalent in the context of talisker.